src/Stratis.Bitcoin.Features.SmartContracts/Rules/SmartContractCoinViewRuleLogic.cs

```diff
@@ -74,9 +79,28 @@ public async Task RunAsync(Func<RuleContext, Task> baseRunAsync, RuleContext con
             uint256 blockRoot = ((ISmartContractBlockHeader)context.ValidationContext.ChainedHeaderToValidate.Previous.Header).HashStateRoot;
 
             this.logger.LogDebug("Block hash state root '{0}'.", blockRoot);
-            byte[] blockRootBytes = blockRoot.ToBytes();
-            this.mutableStateRepository = this.stateRepositoryRoot.GetSnapshotTo(blockRootBytes);
 
+            this.cachedResults = this.executionCache.GetExecutionResult(block.GetHash());
```

Comment:
Not an issue with correctness, but to improve readability could we set this back to `null` after it's finished being used?

Reply:
I think to keep it in-line with the other variables, like `refundCounter` it makes a lot of sense for it to be set only here. Each new run has the responsibility of setting these correctly initially.